### PR TITLE
Add openclip docs

### PR DESCRIPTION
**Pull request by merveenoyan**

### CODE REVIEW

**Comment by merveenoyan** [comment]
Adding light screenshots.

File: docs/hub/open_clip.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Using OpenCLIP at Hugging Face
```
Comment (osanseviero) [inline docs/hub/open_clip.md:1]: I would also add to https://moon-ci-docs.huggingface.co/docs/hub/pr_1131/en/models-libraries

File: docs/hub/open_clip.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+[OpenCLIP](https://github.com/mlfoundations/open_clip) is the open-source library built to reproduce OpenAI's CLIP.
```
Comment (osanseviero) [inline docs/hub/open_clip.md:3]:
```suggestion
[OpenCLIP](https://github.com/mlfoundations/open_clip) is an open-source implementation of OpenAI's CLIP.
```

File: docs/hub/open_clip.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## Exploring OpenCLIP on the Hub
+
+You can find OpenCLIP models by filtering at the left of the [models page](https://huggingface.co/models?library=open_clip&sort=trending).
+
+OpenCLIP models hosted on the Hub have a model card with useful information about the models. Thanks to OpenCLIP Hugging Face Hub integration, you can load OpenCLIP models with few lines of code from the Hub. You can also launch these models using [Inference Endpoints](https://huggingface.co/inference-endpoints).
```
Comment (osanseviero) [inline docs/hub/open_clip.md:9]:
```suggestion
OpenCLIP models hosted on the Hub have a model card with useful information about the models. Thanks to OpenCLIP Hugging Face Hub integration, you can load OpenCLIP models with a few lines of code. You can also deploy these models using [Inference Endpoints](https://huggingface.co/inference-endpoints).
```

File: docs/hub/open_clip.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+## Using existing models
+
+All OpenCLIP models can easily be loaded from the Hub:
+
+```py
+import open_clip
+
+model, preprocess_train, preprocess_val = open_clip.create_model_and_transforms('hf-hub:laion/CLIP-ViT-g-14-laion2B-s12B-b42K')
```
Comment (osanseviero) [inline docs/hub/open_clip.md:28]:
```suggestion
model, preprocess = open_clip.create_model_from_pretrained('hf-hub:laion/CLIP-ViT-g-14-laion2B-s12B-b42K')
```

File: docs/hub/open_clip.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+Once loaded, you can encode the image and text to do zero-shot classification:
+
+```py
+import torch
+from PIL import Image
+
+url = 'http://images.cocodataset.org/val2017/000000039769.jpg'
+image = Image.open(requests.get(url, stream=True).raw)
+image = preprocess_train(image).unsqueeze(0)
```
Comment (osanseviero) [inline docs/hub/open_clip.md:40]:
```suggestion
image = preprocess(image).unsqueeze(0)
```

File: docs/hub/open_clip.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+All OpenCLIP models can easily be loaded from the Hub:
+
+```py
+import open_clip
+
+model, preprocess_train, preprocess_val = open_clip.create_model_and_transforms('hf-hub:laion/CLIP-ViT-g-14-laion2B-s12B-b42K')
+tokenizer = open_clip.get_tokenizer('hf-hub:laion/CLIP-ViT-g-14-laion2B-s12B-b42K')
+```
+
+Once loaded, you can encode the image and text to do zero-shot classification:
```
Comment (osanseviero) [inline docs/hub/open_clip.md:32]: Maybe link to the zero-shot image classification task page here

File: docs/hub/open_clip.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    text_features = model.encode_text(text)
+    image_features /= image_features.norm(dim=-1, keepdim=True)
+    text_features /= text_features.norm(dim=-1, keepdim=True)
+
+    text_probs = (100.0 * image_features @ text_features.T).softmax(dim=-1)
+
+print("Label probs:", text_probs) 
+```
+
+It outputs the following:
```
Comment (osanseviero) [inline docs/hub/open_clip.md:54]:
```suggestion
It outputs the probability of each possible class:
```

**Review by osanseviero — APPROVED** [review]
Very cool! :fire:

File: docs/hub/open_clip.md (hunk 6 above)
Comment (merveenoyan) [reply docs/hub/open_clip.md]: Addressed this.

File: docs/hub/open_clip.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+<div class="flex justify-center">
+<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/openclip_snippet_light.png"/>
+<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/openclip_snippet.png"/>
+</div>
+
+
+## Additional resources
+
+* OpenCLIP [repository](https://github.com/mlfoundations/open_clip)
+* OpenCLIP [docs](https://github.com/mlfoundations/open_clip/tree/main/docs)
```
Comment (pcuenca) [inline docs/hub/open_clip.md:75]:
```suggestion
* OpenCLIP [docs](https://github.com/mlfoundations/open_clip/tree/main/docs)
* OpenCLIP [models in the Hub](https://huggingface.co/models?library=open_clip&sort=trending)
```

**Review by pcuenca — APPROVED** [review]
Awesome!
